Marlin/Conditionals_post.h

```diff
@@ -792,25 +810,41 @@
     #define MANUAL_PROBE_HEIGHT Z_HOMING_HEIGHT
   #endif
 
+  /**
+   * Bed Probing rectangular bounds
+   * These can be further constrained in code for Delta and SCARA
+   */
   #if ENABLED(DELTA)
-    // These will be further constrained in code, but UBL_PROBE_PT values
-    // cannot be compile-time verified within the radius.
-    #define MIN_PROBE_X (-DELTA_PRINTABLE_RADIUS)
-    #define MAX_PROBE_X ( DELTA_PRINTABLE_RADIUS)
-    #define MIN_PROBE_Y (-DELTA_PRINTABLE_RADIUS)
-    #define MAX_PROBE_Y ( DELTA_PRINTABLE_RADIUS)
+    // Probing points may be verified at compile time within the radius
+    // using static_assert(HYPOT2(X2-X1,Y2-Y1)<=DELTA_PRINTABLE_RADIUS,"bad probe point!")
+    // so that may be added to SanityCheck.h in the future.
+    #define MIN_PROBE_X (X_CENTER - (DELTA_PRINTABLE_RADIUS))
+    #define MIN_PROBE_Y (Y_CENTER - (DELTA_PRINTABLE_RADIUS))
+    #define MAX_PROBE_X (X_CENTER +  DELTA_PRINTABLE_RADIUS)
+    #define MAX_PROBE_Y (Y_CENTER +  DELTA_PRINTABLE_RADIUS)
   #elif IS_SCARA
     #define SCARA_PRINTABLE_RADIUS (SCARA_LINKAGE_1 + SCARA_LINKAGE_2)
-    #define MIN_PROBE_X (-SCARA_PRINTABLE_RADIUS)
-    #define MAX_PROBE_X ( SCARA_PRINTABLE_RADIUS)
-    #define MIN_PROBE_Y (-SCARA_PRINTABLE_RADIUS)
-    #define MAX_PROBE_Y ( SCARA_PRINTABLE_RADIUS)
+    #define MIN_PROBE_X (X_CENTER - (SCARA_PRINTABLE_RADIUS))
+    #define MIN_PROBE_Y (Y_CENTER - (SCARA_PRINTABLE_RADIUS))
+    #define MAX_PROBE_X (X_CENTER +  SCARA_PRINTABLE_RADIUS)
+    #define MAX_PROBE_Y (Y_CENTER +  SCARA_PRINTABLE_RADIUS)
   #else
-    // Boundaries for probing based on set limits
-    #define MIN_PROBE_X (max(X_MIN_POS, X_MIN_POS + X_PROBE_OFFSET_FROM_EXTRUDER))
-    #define MAX_PROBE_X (min(X_MAX_POS, X_MAX_POS + X_PROBE_OFFSET_FROM_EXTRUDER))
-    #define MIN_PROBE_Y (max(Y_MIN_POS, Y_MIN_POS + Y_PROBE_OFFSET_FROM_EXTRUDER))
-    #define MAX_PROBE_Y (min(Y_MAX_POS, Y_MAX_POS + Y_PROBE_OFFSET_FROM_EXTRUDER))
+    // Boundaries for Cartesian probing based on set limits
+    #define _MIN_PROBE_X (max(0, X_PROBE_OFFSET_FROM_EXTRUDER))
```

Comment:
We're trying to use the bed limits here, and `X_MIN_POS` has nothing to do with the bed limits. We're going wit the naive assumption that most use 0,0 as the corner or center of the bed.